_api-reference/index-apis/recover.md

```diff
@@ -0,0 +1,289 @@
+
+
+## HTTP and Path methods
+
+```
+GET /_recovery
+GET /<index-name>/recovery/
+```
+
+## Path parameters
+
+Parameter | Type | Description 
+:--- | :--- 
+`index-name` |  String | A comma-separated list of indexes, data streams, or index aliases to which the operation is applied. Supports wildcard expressions (`*`). Use `_all` or `*` to specify all indexes and data streams in a cluster. |
+
+
+## Query parameters
+
+All of the following query parameters are optional.
+
+Parameter | Type | Description 
+:--- | :--- | :---  
+`active_only` | Boolean | When `true`, the response only includes active shard recoveries. Default is `false`.
+`detailed` | Boolean | When `true`, gives detailed information about shard recoveries. Default is `false`.
+`index`  | String | A comma-separated list or wildcard expression of index names used to limit the request.
+
+## Response body
+
+The API responds with the following information about the recovery shard:
+
+Parameter | Type | Description 
+:--- | :--- | :--- 
+`id` | Integer | The ID of the shard. 
+`type` | String | The recovery source for the shard. Returned values include: <br> - `EMPTY_STORE`: An empty store. Indicates a new primary shard or the forced allocation of an empty primary shard using the cluster reroute API. <br> - `EXISTING_STORE`: The store of an existing primary shard. Indicates recovery is related to node startup or the allocation of an existing primary shard. <br> - `LOCAL_SHARDS`: Shards of another index on the same node. Indicates recovery is related to a clone, shrink, or split operation. <br> - `PEER`: A primary shard on another node. Indicates recovery is related to shard replication. <br> - `SNAPSHOT`: A snapshot. Indicates recovery is related to a snapshot restore operation. 
+`STAGE` | String | The recovery stage. Returned values can include: <br> - `INIT`: Recovery has not started. <br> - `INDEX`: Reading index metadata and copying bytes from source to destination. <br> - `VERIFY_INDEX`: Verifying the integrity of the index. <br> - `TRANSLOG`: Replaying transaction log. <br> - `FINALIZE`: Cleanup. <br> - `DONE`: Complete. 
+`primary` | Boolean | When `true`, the shard is a primary shard. 
+`start_time` | String | The timestamp for when the recovery started. 
+`stop_time` | String | The timestamp for when the recovery finished. 
+`total_time_in_millis` | String | The total time to recover shard in milliseconds. 
+`source` | Object | The recovery source. This can include a repository description if recovery is from a snapshot or a description of source node. 
+`target` | Object | The destination node. 
+`index` | Object | The statistics about physical index recovery. 
+`translog` | Object | The statistics about translog recovery. 
+ `start` | Object | The statistics about time to open and start the index. 
+
+## Example requests
+
```

Comment:
The following examples represent how to recover information using the recovery API.